tests/_ast/test_sql_visitor.py

```diff
@@ -626,18 +626,6 @@ def test_find_sql_refs_nested_intersect() -> None:
         """
         assert find_sql_refs(sql) == ["table2", "table3", "table4", "table1"]
 
-    @staticmethod
-    def test_find_sql_refs_with_recursive_cte() -> None:
-        sql = """
-        WITH RECURSIVE cte AS (
-            SELECT 1 AS n FROM table1
-            UNION ALL
-            SELECT n + 1 FROM cte WHERE n < 10
-        )
-        SELECT * FROM cte;
-        """
-        assert find_sql_refs(sql) == ["table1"]
```

Comment:
If this is not yet supported, but you'd like to keep the test, you can decorate it with `pytest.mark.xfail`, as done here: https://github.com/marimo-team/marimo/blob/f8dc9f5c210a44375f5a431771efcfc9a6f17ea7/tests/_ast/test_sql_visitor.py#L569

Reply:
it does work, I didn't see the need for it since we have tested nested cte's. Also noted on the dialect change. I may add back the has_duckdb dependency too.